src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/context/MethodologyContextActionTypes.ts

```diff
@@ -0,0 +1,86 @@
+import { EditableContentBlock } from '@admin/services/publicationService';
+import { ContentSection } from '@common/services/publicationService';
+import { MethodologyContent } from 'src/services/methodology/types';
+import { MethodologyContextState } from './MethodologyContext';
+
+export type ContentSectionKeys = keyof Pick<
+  MethodologyContent,
+  'annexes' | 'content'
+>;
+type BlockMeta = {
+  sectionId: string;
+  blockId: string;
+  sectionKey: ContentSectionKeys;
+};
+type SectionMeta = Omit<BlockMeta, 'blockId'>;
+
+type ClearState = { type: 'CLEAR_STATE' };
+type SetState = { type: 'SET_STATE'; payload: MethodologyContextState };
```

Comment:
Not super keen on this because it kinda breaks the point of using the reducer pattern (might as well just use useState` or `useImmer` directly). Perhaps we could just initialise the reducer with `MethodologyContextState`?

Reply:
I think what you're suggesting would mean that the context is reliant on things happening outside, or before, an instance of the context is rendered.

However, I prefer the reducer to always persist despite what is happening on that page, or with failing API's and for the components/pages using the context to respond to an empty context/loading state/missing data.

Conceptually that makes more sense to me.. I think it's not worth making the changes now, it would also make a case for changing the release reducer too, then updating the tests and I don't think the bang would be worth the buck on this one

Reply:
> I prefer the reducer to always persist despite what is happening on that page, or with failing API's and for the components/pages using the context to respond to a cleared context/loading state.

I'm not quite sure what you mean by that?

It seems like we're just using `SET_STATE` and `CLEAR_STATE` to do an initialisation step for the context, but the context should really be provided with it's initial state from the get-go.

One major reason for doing this is to avoid all of our context consumer components from exhibiting loading flicker whilst the state loads in. A forceful way of doing this would be to just wrap the entire `MethodologyProvider`'s children with a `LoadingSpinner` (or nothing at all) whilst the context's initial state is being fetched. This decision does depend on how loading states are being handled in general across your components, but we do things like this when setting up `AuthContext` for example.

I think it should be possible to make API call in `MethodologyProvider` in`useImmerReducer`'s lazy initialisation parameter. But why even go that far? It looks like we already have the methodology read to go in `MethodologyPage`, so this is not a major change anyway.

Reply:
I think I miss understood your initial message, and didn't realize I was already calling for methodology content in the provider parent